srcjs/shinyapp.js

```diff
@@ -473,8 +472,22 @@ var ShinyApp = function() {
 
   exports.addCustomMessageHandler = addCustomMessageHandler;
 
-  this.dispatchMessage = function(msg) {
-    var msgObj = JSON.parse(msg);
+  this.dispatchMessage = function(data) {
+    var msgObj = {};
+    if(typeof data === "string") {
+      msgObj = JSON.parse(data);
+    } else { // data is arraybuffer
+      var len = new DataView(data,0,1).getUint8(0);
+      var tagdv = new DataView(data,1,len);
```

Comment:
It would be helpful to rename "tag" here to "type", for consistency with the R code.